### DISCUSSION TITLE
New documentation entry

### DISCUSSION THREAD

**Discussion by 77696e63656e74** [JsonMapper/JsonMapper#106] [General]
Hi,
recently I had to register a factory method in the mapper. 
It turned out to be quite simple but this possibility is not documented.
I do not see how I could issue a pull request about it so I'm doing it here, I hope it's okay.
If there is a better approach please let me know.

### Documentation proposition:
**Title**: Factory registration
**Description**:
When using a interface typed property a proper factory must be registered, otherwise JsonMapper will crash. 
**Example**:
```
<?php
$jsonMapper = (new \JsonMapper\JsonMapperFactory())->bestFit();
$factoryRegistry = JsonMapper\Handler\FactoryRegistry::WithNativePhpClassesAdded();
$factoryRegistry->addFactory(
                CarbonInterface::class,
                fn($date) => new Carbon($date)
            );
$jsonMapper->setPropertyMapper(new JsonMapper\Handler\PropertyMapper($factoryRegistry));
```

**Comment by DannyvdSluijs** [comment]
HI @Errancer 

Thanks for your report. If you want the documentation is also available on github => https://github.com/JsonMapper/jsonmapper.github.io
A pull request can be made there

Could you tell more about the crash as Im having some difficulty getting the issue from your report.

**Reply by 77696e63656e74** [reply]
The exception I had was "Unable to resolve un-instantiable Carbon\CarbonInterface as no factory was registered" which is an expected behavior so I wasn't trying to create an issue nor this is a bug report. 
Anyway, thanks for the documentation link, I will make a pull request there.

**Comment by DannyvdSluijs** [comment]
Discussion continued in a pull request on the jsonmapper documentation repo https://github.com/JsonMapper/jsonmapper.github.io/pull/12
